src/main/scala/org/ergoplatform/modifiers/history/Extension.scala

```diff
@@ -0,0 +1,123 @@
+package org.ergoplatform.modifiers.history
+
+import com.google.common.primitives.{Bytes, Shorts}
+import io.circe.Encoder
+import io.circe.syntax._
+import org.ergoplatform.modifiers.BlockSection
+import org.ergoplatform.settings.Algos
+import scorex.core._
+import scorex.core.serialization.Serializer
+import scorex.crypto.authds.LeafData
+import scorex.crypto.hash.Digest32
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+/**
+  * Extension section of Ergo block. Contains two key-value storages,
+  * represented as Seq[(Array[Byte], Array[Byte])] with mandatory and optional fields.
+  *
+  * @param headerId        - id of corresponding header
+  * @param mandatoryFields - fields that are known to all nodes in the network and may be changed
+  *                        via soft/hard forks only. These fields have 4 bytes key and at most `MandatoryFieldValueSize`
+  *                        bytes value.
+  * @param optionalFields  - random data miner may add to a block. This section contains at most `MaxOptionalFields`
+  *                        elements with `OptionalFieldKeySize` byte key size and at most `OptionalFieldValueSize`
+  *                        bytes value size.
+  */
+case class Extension(headerId: ModifierId,
+                     mandatoryFields: Seq[(Array[Byte], Array[Byte])],
```

Comment:
You mean `Array[(Array[Byte], Array[Byte])]`? I'd prefer to have immutable collection here

Reply:
I also hope that `lengthCompare` is overwritten in finite sequences, as scaladoc states: 


> The method as implemented here does not call length directly; its running time is O(length min len) instead of O(length). The method should be overwritten if computing length is cheap.

https://www.scala-lang.org/api/current/scala/collection/SeqLike.html#lengthCompare(len:Int):Int

Reply:
I also think that case class will have the same memory footprint, but will provide a clear field names:
```
case class ExtensionField(key: Array[Byte], value: Array[Byte]) {
  def encodedKey = Algos.encode(key)
 def encodedValue = Algos.encode(value)
}
```